src/server/pps/server/master.go

```diff
@@ -123,14 +135,25 @@ func (a *apiServer) master() {
 	panic("internal error: PPS master has somehow exited. Restarting pod...")
 }
 
-func (a *apiServer) setPipelineFailure(ctx context.Context, specCommit *pfs.Commit, reason string) error {
-	return a.setPipelineState(ctx, specCommit, pps.PipelineState_PIPELINE_FAILURE, reason)
-}
-
 func (a *apiServer) setPipelineCrashing(ctx context.Context, specCommit *pfs.Commit, reason string) error {
 	return a.setPipelineState(ctx, specCommit, pps.PipelineState_PIPELINE_CRASHING, reason)
 }
 
+func (m *ppsMaster) processMonitorUpdates() {
```

Comment:
I made these Monitor thread updates concurrent here. In addition to a tad more concurrency, I think having modifications to Monitors encapsulated in the master helps keep the boundary of responsibilities cleaner between the Master goro, and the pipelineOp goros